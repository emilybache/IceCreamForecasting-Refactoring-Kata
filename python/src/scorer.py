import enum
import random


class IceCream(enum.Enum):
    Strawberry = 1
    Chocolate = 2
    Vanilla = 3


flavour = None


def get_score():
    sunny_today = lookup_weather()
    global flavour
    if flavour == IceCream.Strawberry:
        if sunny_today:
            return 10
        else:
            return 5
    elif flavour == IceCream.Chocolate:
        return 6
    elif flavour == IceCream.Vanilla:
        if sunny_today:
            return 7
        else:
            return 5
    else:
        return -1


def lookup_weather():
    # placeholder implementation - real version would make API call to weather service
    sunny = random.randrange(0, 1)
    return bool(sunny)


def update_selection():
    # placeholder implementation - real version would use machine learning to predict sales
    global flavour
    flavour = random.choice([IceCream.Strawberry, IceCream.Chocolate, IceCream.Vanilla])
