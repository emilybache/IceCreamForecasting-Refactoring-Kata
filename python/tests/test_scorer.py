import approvaltests
import pytest
import requests

import scorer
from scorer import get_score, IceCream, get_score_with_weather_and_flavour


def print_get_score(args, result):
    return f"{args} => {result}\n"


def test_get_score_with_weather():
    weathers = [True, False]
    flavours = [IceCream.Strawberry, IceCream.Chocolate, IceCream.Vanilla, None]
    approvaltests.verify_all_combinations(
        get_score_with_weather_and_flavour,
        [
            weathers,
            flavours,
        ],
        formatter=print_get_score
    )


class StubWeatherServiceResponse:
    def __init__(self):
        self.status_code = 200

    def json(self):
        return {"weather": {"main": "Sunny"}}


def test_lookup_weather_default_location(monkeypatch):
    def stub_requests_get(*args, **kwargs):
        return StubWeatherServiceResponse()

    monkeypatch.setattr(requests, "get", stub_requests_get)
    assert scorer.lookup_weather() == True


@pytest.mark.vcr
def test_lookup_weather_not_sunny():
    location_aare = (63.3990, 13.0815)
    assert scorer.lookup_weather(location_aare) == False


@pytest.mark.skip("this test is too fragile to be worth maintaining")
def test_get_score(monkeypatch):
    def stub_lookup_weather():
        return True

    monkeypatch.setattr(scorer, "lookup_weather", stub_lookup_weather)
    monkeypatch.setattr(scorer, "flavour", IceCream.Vanilla)

    def mock_get_score_with_weather(sunny_today, flavour):
        assert sunny_today == True
        assert flavour == IceCream.Vanilla
        return 0

    monkeypatch.setattr(scorer, "get_score_with_weather", mock_get_score_with_weather)
    assert scorer.get_score() == 0
