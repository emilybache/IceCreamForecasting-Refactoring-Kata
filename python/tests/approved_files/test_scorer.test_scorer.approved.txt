(<IceCream.Strawberry: 1>, True) 	=> 	score=10
(<IceCream.Strawberry: 1>, False) 	=> 	score=5
(<IceCream.Chocolate: 2>, True) 	=> 	score=6
(<IceCream.Chocolate: 2>, False) 	=> 	score=6
(<IceCream.Vanilla: 3>, True) 	=> 	score=7
(<IceCream.Vanilla: 3>, False) 	=> 	score=5
